scripts/build.py

```diff
@@ -549,6 +525,38 @@ def save_hashes_as_json(target_filepath, file_hashes):
     file_hashes[relative_filepath] = file_hash
 
 
+def minify_func(source_path, target_path, file_hashes, filename):
+    if filename.endswith('.html'):
+        process_html(source_path, target_path, file_hashes)
+    elif filename.endswith('.css'):
+        _minify(source_path, target_path)
+    elif filename.endswith('.js'):
+        _minify(source_path, target_path)
+    else:
+        shutil.copyfile(source_path, target_path)
+
+
+def _execute_tasks(tasks, batch_size=24):
+    """Starts all tasks and checks the results.
+
+    Runs no more than 'batch_size' tasks at a time.
+    """
+    remaining_tasks = [] + tasks
+    currently_running_tasks = set([])
+
+    while remaining_tasks or currently_running_tasks:
+        if currently_running_tasks:
+            for task in list(currently_running_tasks):
+                task.join(1)
+                if not task.isAlive():
+                    currently_running_tasks.remove(task)
+
+        while remaining_tasks and len(currently_running_tasks) < batch_size:
+            task = remaining_tasks.pop()
+            currently_running_tasks.add(task)
+            task.start()
```

Comment:
You should have a sleep operation (for about a second) after this, otherwise this will continually loop and will probably take over a thread.

Reply:
I'm not sure I understand what is meant by take over a thread. Why is the sleep necessary?

Reply:
This is so that you can free up resources to allow for other processes to take place, otherwise it'll constantly check whether other minification threads are done.